src/firmware/main/app/primitives/shoot_primitive.c

```diff
@@ -0,0 +1,173 @@
+#include "firmware/main/app/primitives/shoot_primitive.h"
+
+#include <math.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "firmware/main/app/control/bangbang.h"
+#include "firmware/main/app/control/control.h"
+#include "firmware/main/app/control/physbot.h"
+#include "firmware/main/shared/physics.h"
+#include "firmware/main/shared/util.h"
+
+// these are set to decouple the 3 axis from each other
+// the idea is to clamp the maximum velocity and acceleration
+// so that the axes would never have to compete for resources
+#define TIME_HORIZON 0.05f  // s
```

Comment:
Same as above. Comments and the define seem unrelated.

Reply:
Easy fix, so I'll change this one. Nuked the comment.